### ISSUE TITLE
[Bug]: Error changing data: cannot store TEXT value in INT column

### ISSUE THREAD

**Issue by dws14159** [sqlitebrowser/sqlitebrowser#3805]
### What did you do?

I have a STRICT table with a NULLable INT column that I'm modifying in the Browse Data tab.

### What did you expect to see?

If I Tab out of this column without entering anything, I'd expect it to accept the blank input as meaning NULL and not throw an error about it.

Looking at the log this error makes sense. But I think the generated SQL should be something like '... SET "FromAccount" = NULL ...'

### What did you see instead?

I get a popup message "Error changing data: cannot store TEXT value in INT column".

![image](https://github.com/user-attachments/assets/9efbff20-b928-473d-8061-aca0eb2b4a13)

### DB4S Version

3.13.1

### What OS are you seeing the problem on?

Windows

### OS version

Windows 11 Pro 10.0.22631

### Relevant log output

```shell
PRAGMA foreign_keys = '1';
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
PRAGMA encoding;
SELECT "_rowid_",* FROM "main"."Accounts" LIMIT 49999 OFFSET 0;
SELECT "_rowid_",* FROM "main"."Transactions" LIMIT 49999 OFFSET 0;
SAVEPOINT "UNDOPOINT";
SAVEPOINT "RESTOREPOINT";
UPDATE "main"."Transactions" SET "FromAccount"='' WHERE "_rowid_"='2'
SAVEPOINT "UNDOPOINT";
UPDATE "main"."Transactions" SET "FromCategory"='' WHERE "_rowid_"='2'
```

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by justinclift** [comment]
Yeah, that does sound like something we should make sure works with (nullable) non-TEXT columns. :smile:

**Comment by mgrojo** [comment]
As a workaround, you can press Alt+Del to set the cell to null.

By the way, I don't reproduce it just by tabbing. I had to enter a number, then edit the cell and set it to empty to reproduce the issue. On this regard, it doesn't seem that important.

**Comment by justinclift** [comment]
Ahhh.  Yeah that sounds like something a lot less common than it sounded. :smile:

**Comment by dws14159** [comment]
I don't enter any data. I've reproduced the problem with a simpler test table:

```
CREATE TABLE "ErrorTest" (
	"Id"	INTEGER,
	"Field2"	INT,
	"Field3"	INT,
	"Field4"	TEXT,
	PRIMARY KEY("Id")
) STRICT;
```

I then go into Browse Data, select this table, click "Insert a new record", click on Id (then click again to make it editable), then Tab into Field2. Then I hit Tab again and the error immediately pops up. Dismiss it and the cursor is now in Field3. Tab again, still not entering any data, and the error pops up again. Alt-Del doesn't prevent the error (also tried Alt-Backspace, Alt Gr-Backspace and Alr Gr-Del).

**Comment by dws14159** [comment]
Made a quick recording using the OSK:

https://github.com/user-attachments/assets/cb848919-8038-484c-b666-6fdec9b6ecd0

**Comment by mgrojo** [comment]
Thanks for the detailed steps. It's clear now what it is happening.

The workaround in that case would be to press ESC to finish the edition of the field that has to keep the NULL value, then Tab to continue to the next field and F2 again to enter the edit mode.

In case someone wants to fix this, the correct solution should be to make the `EditDelegate` in `AddRecordDialog` to be available for `ExtendedTableWidget`. That line editor implementation already supports entering null values with the shortcut, and it will keep an existing NULL value until the user types anything else.
